core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -513,6 +515,38 @@ protected long snapshotId() {
     return snapshotId;
   }
 
+  protected static <F extends ContentFile<F>> List<ManifestFile> writeFilesToManifests(
```

Comment:
I feel like , its easier to read and fits more Iceberg pattern if the logic is inside ManifestWriter (similar to RollingFileWriter), is that possible?  Can also eliminate the double method for data/delete file.

Reply:
It seems possible. Let me try to implement it.

Reply:
Thanks @ConeyLiu !